firmware/library/L1_Drivers/ssp.hpp

```diff
@@ -70,3 +72,215 @@ class SspInterface
+        data_size_(kEight), frame_format_(kSpi), clock_polarity_(1),
+        clock_phase_(0), clock_divider_(0), clock_prescaler_(2)
+    {
+    }
+
+    // User modified constructor. **MUST** be followed by the setter functions
+    // for Power, Clock, and Frame format found below.
+    explicit constexpr Ssp(Peripheral set_pssp) :
+        mosi_(mosi_pin_),
+        miso_(miso_pin_),
+        sck_(sck_pin_),
+
+        mosi_pin_(kPort0, ((set_pssp == Peripheral::kSsp0) ? kPin18 : kPin9)),
+        miso_pin_(kPort0, ((set_pssp == Peripheral::kSsp0) ? kPin17 : kPin8)),
+        sck_pin_(kPort0, ((set_pssp == Peripheral::kSsp0) ? kPin15 : kPin7)),
+
+        // Must set all these values, default to 0
+        pssp_(set_pssp), pconp_bits_(static_cast<PowerOn>(0)),
+        master_mode_(static_cast<MasterSlaveMode>(0)),
+        data_size_(static_cast<DataSize>(0)),
+        frame_format_(static_cast<FrameMode>(0)), clock_polarity_(0),
+        clock_phase_(0), clock_divider_(0), clock_prescaler_(0)
+    {
+    }
+
+    // Unit-testing constructor, removed for production
+    constexpr Ssp(Peripheral pssp,
+
+        PinConfigureInterface & mosi_pin,
+        PinConfigureInterface & miso_pin,
+        PinConfigureInterface & sck_pin) :
+
+        mosi_(mosi_pin),
+        miso_(miso_pin),
+        sck_(sck_pin),
+
+        mosi_pin_(PinConfigure::CreateInactivePin()),
+        miso_pin_(PinConfigure::CreateInactivePin()),
+        sck_pin_(PinConfigure::CreateInactivePin()),
+
+        pssp_(pssp), pconp_bits_(kPconp0), master_mode_(kMaster),
+        data_size_(kEight), frame_format_(kSpi), clock_polarity_(1),
+        clock_phase_(0), clock_divider_(0), clock_prescaler_(2)
+    {
+    }
+
+    // Powers on the desired peripheral and sets the peripheral clock. Also,
+    // enables the SSP data lines, format, data size, and more. See page 601
+    // of user manual UM10562 LPC408x/407x for more details.
+    bool Initialize() override
+    {
+        bool peripheral_initialized;
+
+        if (pssp_ > Peripheral::kSsp1)
+        {
+            peripheral_initialized = 0;
+        }
+        else
+        {
+            // if frame mode is set for Microwire, force data_size_ = 8 bits
+            DataSize check_data_size = (frame_format_ == kMicro)
+                                        ? kEight : data_size_;
+            // Power up peripheral
+            sysclock_register->PCONP &= ~(1 << pconp_bits_);
+            sysclock_register->PCONP |= (1 << pconp_bits_);
+            ssp_registers[static_cast<uint32_t>(pssp_)]->CPSR &= ~(0xFF);
+            ssp_registers[static_cast<uint32_t>(pssp_)]->CPSR |=
+                (clock_prescaler_);
+
+            // Enable SSP pins
+            mosi_.SetPinFunction(kEnableSspPins);
+            miso_.SetPinFunction(kEnableSspPins);
+            sck_.SetPinFunction(kEnableSspPins);
+
+            // Clear and Set Control Register values
+            ssp_registers[static_cast<uint32_t>(pssp_)]->CR0 &=

```

Comment:
Rather than having to write `static_cast` multiple times, create a local variable.